lib/system_commands.rb

```diff
@@ -32,8 +33,24 @@ def sh!(*args)
       end
     end
 
-    output = `#{cmd} 2>&1`
-    status = $?
+    output = nil
+    status = nil
+
+    if options[:timeout] > 0
+      cmd = "timeout #{options[:timeout]+1} #{cmd}"
+      begin
+        Timeout::timeout(options[:timeout]) do
```

Comment:
Can we get rid of this (and the +1 thing above?) since the `timeout` command should fail (with exit status 124) when time runs out.